biocrnpyler/construct_explorer.py

```diff
@@ -165,8 +164,14 @@ def terminate_loop(self):
         self.initialize_loop()
 
     #Returns a list of RNAconstructs
-    def return_components(self,component):
-        return self.all_rnas.values()
+    def return_components(self,component, previously_enumerated = None):
```

Comment:
the component enumerators not only return which components would be created but they also populate components within the input construct with proper values. For example an "RBS" has a "protein" member, which needs to be populated. But for some reason this population is not happening as the thing is getting enumerated but only after it is done. I think functionally it is the same. I cannot easily get rid of having seperate "return_components" for TxExplorer and TlExplorer because they have accumulator variables named different things.

If you feel strongly about this I could change it so that they are more similar and maybe get rid of having this different function. But it would be a bunch of work for not a lot of benefit in my opinion